store/src/java/com/zimbra/cs/service/admin/SetCurrentVolume.java

```diff
@@ -49,12 +50,16 @@ private SetCurrentVolumeResponse handle(SetCurrentVolumeRequest req, Map<String,
         short volId = req.getId() > 0 ? req.getId() : Volume.ID_NONE;
         VolumeManager.getInstance().setCurrentVolume(req.getType(), volId);
         SetCurrentVolumeResponse response = new SetCurrentVolumeResponse();
-        Volume volume = VolumeManager.getInstance().getVolume(volId);
-        // if its primary volume then
-        if (Volume.TYPE_MESSAGE == volume.getType()) {
-            // set current store manager
-            StoreManagerRuntimeSwitchResult runtimeSwitchResult = StoreManagerResetHelper.setNewStoreManager(volume.getStoreManagerClass());
-            response.setRuntimeSwitchResult(runtimeSwitchResult);
+        if (volId != Volume.ID_NONE) {
+            Volume volume = VolumeManager.getInstance().getVolume(volId);
+            // if its primary volume then
+            if (Volume.TYPE_MESSAGE == volume.getType()) {
```

Comment:
what happens if this is true?

Reply:
There is no need to perform any action here for secondary volume after this condition. This is existing behavior so we didn't make any changes.